### ISSUE TITLE
consider adding github action support

### ISSUE THREAD

**Issue by danmarsden** [Vidalia/moodle-format_cards#2]
Some of the tests run by the Moodle.org plugins db can be run via github actions on each commit in your github repo. Enabling this helps you to make sure future changes to your plugin will continue to follow the guidelines.

the short version - grab this file:
https://github.com/moodlehq/moodle-plugin-ci/blob/master/gha.dist.yml
rename it as ci.yml and put into the directory .github/workflows within tyour project eg:
https://github.com/danmarsden/moodle-mod_attendance/blob/MOODLE_311_STABLE/.github/workflows/ci.yml

then on every commit you make to github it will fire off a request to run the tests and will give you traffic lights beside each commit and generate a report.

note - this is not a blocker for plugins db approval

**Comment by Vidalia** [comment]
I've added in the actions support, seemed too easy not to. It seems to be picking up some extra items that the phpcs rules on my machine is missing but I expect I've misconfigured my local environment wrong somehow.

The stylelint:css rules don't seem to understand the `aspect-ratio` property, [which is widely supported](https://caniuse.com/mdn-css_properties_aspect-ratio).

If It's not a blocker I'll try to address everything else later. None of the warnings it's spitting out look particularly concerning
